src/DynamoCoreWpf/ViewModels/PackageManager/PublishPackageViewModel.cs

```diff
@@ -881,44 +882,67 @@ public static PublishPackageViewModel FromLocalPackage(DynamoViewModel dynamoVie
             {
-                Group = l.Group,
-                Description = l.Description,
-                Keywords = l.Keywords != null ? String.Join(" ", l.Keywords) : "",
+                Group = pkg.Group,
+                Description = pkg.Description,
+                Keywords = pkg.Keywords != null ? String.Join(" ", pkg.Keywords) : "",
                 CustomNodeDefinitions = defs,
-                Name = l.Name,
-                RepositoryUrl = l.RepositoryUrl ?? "",
-                SiteUrl = l.SiteUrl ?? "",
-                Package = l,
-                License = l.License,
-                SelectedHosts = l.HostDependencies as List<string>,
-                CopyrightHolder = l.CopyrightHolder,
-                CopyrightYear = l.CopyrightYear
+                Name = pkg.Name,
+                RepositoryUrl = pkg.RepositoryUrl ?? "",
+                SiteUrl = pkg.SiteUrl ?? "",
+                Package = pkg,
+                License = pkg.License,
+                SelectedHosts = pkg.HostDependencies as List<string>,
+                CopyrightHolder = pkg.CopyrightHolder,
+                CopyrightYear = pkg.CopyrightYear
             };
 
             // add additional files
-            l.EnumerateAdditionalFiles();
-            foreach (var file in l.AdditionalFiles)
+            pkg.EnumerateAdditionalFiles();
+            foreach (var file in pkg.AdditionalFiles)
             {
-                vm.AdditionalFiles.Add(file.Model.FullName);
+                pkgViewModel.AdditionalFiles.Add(file.Model.FullName);
             }
 
-            var nodeLibraryNames = l.Header.node_libraries;
+            var nodeLibraryNames = pkg.Header.node_libraries;
 
             var assembliesLoadedTwice = new List<string>();
             // load assemblies into reflection only context
-            foreach (var file in l.EnumerateAssemblyFilesInBinDirectory())
+
+            //TODO unclear how many MLC there should be - if we want to recreate the behavior from before -
+            //maybe only 1? For now one per package publish.
+
+            // Retrieve the location of the assembly and the referenced assemblies used by the domain
+            var runtimeAssemblies = Directory.GetFiles(RuntimeEnvironment.GetRuntimeDirectory(), "*.dll");
+            // Create the list of assembly paths consisting of runtime assemblies.
+            var assemblyPaths = new List<string>(runtimeAssemblies);
+
+            //and all the package assemblies.
+            foreach (var assemblyFile in pkg.EnumerateAssemblyFilesInPackage())
+            {
+                assemblyPaths.Add(assemblyFile);
+            }
+   
+            // Create PathAssemblyResolver that can resolve assemblies using the created list.
+            var resolver = new PathAssemblyResolver(assemblyPaths);
+            var mlc = new MetadataLoadContext(resolver);
+
+            //TODO for now, we leak the mlc, but it's unlikely users will publish
```

Comment:
how are we leaking the mlc ?
is that why we are not cleaning up the mlc ?
There might be a lot of assemblies loaded from the runtime libs....are you saying it is not probable that a user publishes multiple times in the same session ?

Reply:
yes, I am not disposing the mlc here - its assemblies seem to live as package dependencies for quite some time. If we moved to only using one MLC per session I think there would be a smaller chance of any appreciable memory leak - ie it would be more like using the ReflectionLoadContext.

Yes I think it's unlikely that a user publishes multiple times in a single session.

At this point it's not clear to me that there is ever a good time to dispose these mlcs.

What do you think?

Reply:
What happens when you dispose the mlc at the end of this function? Is it holding onto assembly references that are also owned by the package?

Reply:
The assemblies loaded in the mlc seem to be stored in the package view model
As Mike said, it is not obvious how long they are kept around

Reply:
If we try to use a single mlc, would that increase the chance of conflicts that were not there with reflectionOnly ?
Also, seems like mlc is a little more permissive than reflectionOnly. As an example I think BadImageFormatException exceptions will not eever be thrown.

Reply:
We should test this feature as much as possible (in revit and sandbox) since we do not know all the implications yet

Reply:
@pinzart90 I've tested your concern, and even though the docs don't mention it, it throws the same exception when loading a native assembly.
<img width="769" alt="Screenshot 2023-11-08 at 9 19 13 PM" src="https://github.com/DynamoDS/Dynamo/assets/508936/cadd0678-337b-4cf4-ba2c-2279c3afee7d">

Also I have written a test for it.